File: tools/poyo/cmd/route_remove.go
package cmd

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"

	"poyo-cli/internal/config"
	"poyo-cli/internal/fsutil"
	"poyo-cli/internal/routes"
	"poyo-cli/internal/tui"

	"github.com/spf13/cobra"
)

var removeCmd = &cobra.Command{
	Use:   "remove <path>",
	Short: "Remove a route",
	Args:  cobra.ExactArgs(1),
	RunE:  runRemove,
}

func init() {
	routeCmd.AddCommand(removeCmd)
}

func runRemove(cmd *cobra.Command, args []string) error {
	urlPath := args[0]
	r, err := routes.Read(config.RoutesJSON)
	if err != nil {
		return err
	}

	idx := -1
	for i := range r {
		if strings.EqualFold(r[i].Path, urlPath) || strings.EqualFold(r[i].Path, "/"+strings.TrimPrefix(urlPath, "/")) {
			idx = i
			break
		}
	}

	if idx == -1 {
		return fmt.Errorf("route not found: %s", urlPath)
	}

	routeToRemove := r[idx]
	controller := routeToRemove.Controller
	
	deleteController := false
	if controller != "" {
		cPath := filepath.Join(config.ControllersDir, controller+".cs")
		if _, err := os.Stat(cPath); err == nil {
			q := fmt.Sprintf("Route uses custom controller '%s'. Delete this controller file?", controller)
			confirmed, err := tui.Confirm(q)
			if err != nil {
				return err
			}
			deleteController = confirmed
		}
	}

	deleteFilesQ := "Do you want to DELETE the physical files and folders related to this route?"
	deleteFiles, err := tui.Confirm(deleteFilesQ)
	if err != nil {
		return err
	}

	// Logic Execution
	if deleteController {
		cPath := filepath.Join(config.ControllersDir, controller+".cs")
		os.Remove(cPath)
		fmt.Printf("[DELETED] Controller: %s.cs\n", controller)
	}

	if deleteFiles {
		reactPath := filepath.Join(config.ClientDir, routeToRemove.Files.React)
		viewPath := filepath.Join(config.ServerDir, routeToRemove.Files.View)

		if _, err := os.Stat(reactPath); err == nil {
			os.Remove(reactPath)
			fmt.Printf("[DELETED] React Page: %s\n", routeToRemove.Files.React)
			fsutil.DeleteEmptyParents(reactPath, config.ClientDir)
		}

		if _, err := os.Stat(viewPath); err == nil {
			os.Remove(viewPath)
			fmt.Printf("[DELETED] MVC View: %s\n", routeToRemove.Files.View)
			fsutil.DeleteEmptyParents(viewPath, config.ServerDir)
		}
	}

	// Remove from list
	r = append(r[:idx], r[idx+1:]...)
	if err := routes.Write(config.RoutesJSON, r); err != nil {
		return err
	}

	fmt.Printf("[REMOVED] Route '%s' removed from routes.json\n", routeToRemove.Path)

	if !deleteFiles {
		fmt.Println("[INFO] Orphaned files (not deleted):")
		fmt.Printf("  - poyo.client/%s\n", routeToRemove.Files.React)
		fmt.Printf("  - Poyo.Server/%s\n", routeToRemove.Files.View)
	}

	return nil
}

File: packages/poyo/src/commands/route-remove.ts
import fs from "node:fs";
import { Command } from "commander";
import { getPaths } from "../config.js";
import { CliError } from "../error.js";
import { findRoute } from "../registry.js";
import { readRoutes, writeRoutes } from "../registry.js";
import { writeRouteManifest } from "../route-manifest.js";
import { deleteEmptyParents } from "../scaffold.js";

export function removeCommand(): Command {
	return new Command("remove")
		.description("Remove a route from routes.json")
		.argument("<path>", "URL path of the route to remove")
		.option(
			"-y, --yes",
			"Skip confirmation and delete related files automatically",
		)
		.option(
			"--keep-files",
			"Remove the route from routes.json but keep its files on disk",
		)
		.action(async (urlPath: string, options) => {
			const paths = getPaths();
			const routes = readRoutes(paths);
			const route = findRoute(routes, urlPath);
			if (!route) {
				throw new CliError(`Route not found: ${urlPath}`);
			}

			const hasController = route.controller !== undefined;

			let deleteController = false;
			let deleteFiles = false;

			if (options.yes) {
				deleteController = hasController;
				deleteFiles = true;
			} else if (options.keepFiles) {
				deleteController = false;
				deleteFiles = false;
			} else {
				const { confirm } = await import("@inquirer/prompts");
				if (hasController) {
					deleteController = await confirm({
						message: `Route uses custom controller '${route.controller}'. Delete this controller file?`,
						default: false,
					});
				}
				deleteFiles = await confirm({
					message:
						"Do you want to DELETE the physical files and folders related to this route?",
					default: false,
				});
			}

			if (deleteController && route.controller) {
				const controllerPath = `${paths.controllersDir}/${route.controller}.cs`;
				if (fs.existsSync(controllerPath)) {
					fs.unlinkSync(controllerPath);
				}
			}

			if (deleteFiles) {
				const reactPath = `${paths.clientDir}/${route.files.react}`;
				const viewPath = `${paths.serverDir}/${route.files.view}`;
				if (fs.existsSync(reactPath)) {
					fs.unlinkSync(reactPath);
					deleteEmptyParents(reactPath, paths.clientDir);
				}
				if (fs.existsSync(viewPath)) {
					fs.unlinkSync(viewPath);
					deleteEmptyParents(viewPath, paths.serverDir);
				}
			}

			const index = routes.findIndex((r) => r.path === route.path);
			routes.splice(index, 1);
			writeRoutes(paths, routes);
			writeRouteManifest(paths);

			if (!deleteFiles) {
				process.stdout.write(
					`Orphaned files kept (use 'poyo route remove ${route.path} --yes' to delete them):\n`,
				);
				process.stdout.write(
					`  - ${paths.toProjectPath("client", route.files.react)}\n`,
				);
				process.stdout.write(
					`  - ${paths.toProjectPath("server", route.files.view)}\n`,
				);
				if (deleteController === false && route.controller) {
					process.stdout.write(
						`  - ${paths.toProjectPath("server", `Controllers/${route.controller}.cs`)}\n`,
					);
				}
			}
		});
}
